openpype/hosts/traypublisher/api/editorial.py

```diff
@@ -0,0 +1,242 @@
+import re
+from copy import deepcopy
+
+from openpype.client import get_asset_by_id
+from openpype.pipeline.create import CreatorError
+
+
+class ShotMetadataSover:
+    """Collecting hierarchy context from `parents` and `hierarchy` data
+    present in `clip` family instances coming from the request json data file
+
+    It will add `hierarchical_context` into each instance for integrate
+    plugins to be able to create needed parents for the context if they
+    don't exist yet
+    """
+
+    NO_DECOR_PATERN = re.compile(r"\{([a-z]*?)\}")
+
+    # presets
+    clip_name_tokenizer = None
+    shot_rename = True
+    shot_hierarchy = None
+    shot_add_tasks = None
+
+    def __init__(self, creator_settings, logger):
```

Comment:
Why not explicitly specify what arguments must be passed instead of dictionary with expected keys?